gravitee-apim-console-webui/src/management/api/component/plan/api-plan-form.component.ts

```diff
@@ -514,41 +528,44 @@ const internalFormValueToPlanV4 = (
   mode: 'create' | 'edit',
   planFormType: PlanFormType,
   apiType: ApiType,
+  hasTcpListeners: boolean,
 ): PlanFormValue => {
   // Init flows with restriction step. Only used in create mode
   const initFlowsWithRestriction = (restriction: InternalPlanFormValue['restriction']): FlowV4[] => {
-    const restrictionPolicies: StepV4[] = [
-      ...(restriction.rateLimitEnabled
-        ? [
-            {
-              enabled: true,
-              name: 'Rate Limiting',
-              configuration: restriction.rateLimitConfig,
-              policy: 'rate-limit',
-            },
-          ]
-        : []),
-      ...(restriction.quotaEnabled
-        ? [
-            {
-              enabled: true,
-              name: 'Quota',
-              configuration: restriction.quotaConfig,
-              policy: 'quota',
-            },
-          ]
-        : []),
-      ...(restriction.resourceFilteringEnabled
-        ? [
-            {
-              enabled: true,
-              name: 'Resource Filtering',
-              configuration: restriction.resourceFilteringConfig,
-              policy: 'resource-filtering',
-            },
-          ]
-        : []),
-    ];
+    const restrictionPolicies: StepV4[] = hasTcpListeners
```

Comment:
This is necessary because if the Step is not available, no restrictions can be added ? 

Otherwise, I'd call it an enableRestriction so that it's linked to the function rather than the "context". As for DisplaySecurityStep